lib/autoyast.rb

```diff
@@ -262,26 +262,25 @@ def apply_url_extraction(xml)
     end
   end
 
-  def apply_extracted_files(scope)
-    return if !@system_description[scope] || !@system_description[scope].extracted
-
-    base = Pathname(@system_description.scope_file_store(scope).path)
-    Dir["#{base}/**/*"].sort.each do |path|
-      next if File.directory?(path)
-
-      relative_path = Pathname(path).relative_path_from(base).to_s
-      url = "`cat /tmp/description_url`/#{URI.escape(File.join(scope, relative_path))}"
-
-      @chroot_scripts << "mkdir -p '#{File.join("/mnt", File.dirname(relative_path))}'"
-      @chroot_scripts << "curl -o '#{File.join("/mnt", relative_path)}' \"#{url}\""
-    end
-
-    @system_description[scope].files.map do |file|
-      if file.user
-        @chroot_scripts << "chown #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
-      end
-      if file.mode
-        @chroot_scripts << "chmod #{file.mode} '#{File.join("/mnt", file.name)}'"
+  def apply_changed_files(scope)
+    return if !@system_description.scope_extracted?(scope)
+
+    @system_description[scope].files.each do |file|
+      if file.deleted?
+        @chroot_scripts << "rm -rf '#{Machinery.escape_single_quotes(file.name)}'"
+      elsif file.directory?
+        @chroot_scripts << "chmod #{file.mode} '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
+        @chroot_scripts << "chown #{file.user}:#{file.group} '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
+      elsif file.file?
+        url = "`cat /tmp/description_url`/#{URI.escape(File.join(scope, file.name))}"
+        @chroot_scripts << "mkdir -p '#{Machinery.escape_single_quotes(File.join("/mnt", File.dirname(file.name)))}'"
+        @chroot_scripts << "curl -o '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}' \"#{url}\""
+        @chroot_scripts << "chmod #{file.mode} '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
+        @chroot_scripts << "chown #{file.user}:#{file.group} '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
+      elsif file.link?
+        @chroot_scripts << "rm -rf '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
+        @chroot_scripts << "ln -s '#{Machinery.escape_single_quotes(file.target)}' '#{Machinery.escape_single_quotes(File.join("/mnt", file.name))}'"
```

Comment:
Line is too long. [149/100]<br>Prefer single-quoted strings inside interpolations.